Fix double URL encoding

yecnj opened the pull request:
### Description



#### Changelog
- Fixed an issue where URLs were being encoded twice
- Corrected handling of the `src/pages/` path in server configuration



#### Process
In `+server.js`, the URL was being encoded twice due to the following two steps:
- Use of `new URL` to support Windows file paths
- Calculation of `node.href` in the `_buildPageManifest` function.

The following log from `Sidebar.svelte` illustrates the double encoding issue with the fileTree object:
```bash
'white%20space': {
  label: 'white%20space',
  href: '/white%2520space',
  children: {},
  frontMatter: [Object],
  isTemplated: false,
  isPage: true
}
```
![image](https://github.com/user-attachments/assets/329dcce1-494d-4df4-bb7d-51d9d9168c3b)

The `_buildPageManifest` function expects an unencoded `pagePath` for the computation of variables such as `part` and `label`.

To rectify this, I replaced the use of `new URL` with `path.normalize` and `replace` to maintain Windows compatibility.

This change also led me to adjust how paths starting with `/src/pages` are handled, changing them to start with `src/pages` instead.

Here are the updated results logged from the `Sidebar.svelte` file showing the corrected handling:

```bash
'white space': {
  label: 'white space',
  href: '/white%20space',
  children: {},
  frontMatter: [Object],
  isTemplated: false,
  isPage: true
}
```

![image](https://github.com/user-attachments/assets/f0c22feb-3fd2-4e7b-9aab-4b123d50c142)

This resolves issue #2953 .



#### Additional Comment
When I first started reading at the code, I thought `example-project` was a test repository unrelated to deployment.

However, I noticed that it's actually a "template" being copied to `.evidence/template`, based on various PRs such as #1737 and #2529. Have I understood this correctly?

If so, to avoid confusion, how about you rename `example-project` to something like `project-template`?




### Checklist

- [x] For UI or styling changes, I have added a screenshot or gif showing before & after
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)

archiewood commented:
This appears to fail the e2e tests on windows

archiewood commented on sites/example-project/src/pages/api/pagesManifest.json/+server.js:
```diff
@@ -71,7 +71,8 @@ export async function GET() {
 				const relative_path = path.join(dirent.parentPath ?? dirent.path ?? dir, dirent.name);
 				const content = await fs.readFile(relative_path, 'utf-8');
 				// regularize for windows
-				pages[new URL(`file:///${relative_path}`).pathname] = content;
```
Note the comment above this removed line
```
// regularize for windows
```

yecnj commented:
Thanks for catching this! I'll test this change to verify it resolves the issue.

yecnj commented:
As mentioned in the PR description, using `new URL` causes double encoding issues.
I've modified the code to achieve similar functionality using `path.sep` and `join`, and reverted the path processing method back to its original form as the current approach was prone to issues.

I'll wait for workflow approval to proceed with the deployment.

archiewood commented:
@ItsMeBrianD can you kick CI again please

archiewood commented:
@yecnj - It seems this latest is causing the docs deploy to fail. The docs use Evidence, so this makes it a decent chance that this has broken some core Evidence functionality. 

If you are developing, running the docs `pnpm run sources:docs && pnpm run dev:docs` is a decent way to check if your changes are working

Error message on netlify:

https://app.netlify.com/sites/next-docs-evidence/deploys/677b31ce721c750008d7a5f5

archiewood reviewed (changes requested):
appears to be causing docs to fail to build

yecnj commented:
@archiewood Thanks for the thorough review! While I couldn't reproduce any issues in local builds, I've added the forward slash prefix to ensure consistent path handling.

 I've also checked that the Netlify build is now passing successfully.

https://app.netlify.com/sites/next-docs-evidence/deploys/677c8f22f610230008e921dd

yecnj commented:
hmm. I will check windows test.

archiewood commented:
it might just be a flaky test, let me retry it

yecnj commented:
I think something is pending now.
I tried log-in to chromatic but didn't get much results.
can you do me a favor?

archiewood commented:
Looks good in CI. @ItsMeBrianD any final thought or good to merge ?

archiewood commented:
@yecnj thank you for your contribution - this will go out during our next release
